pyccel/parser/semantic.py

```diff
@@ -2543,6 +2547,11 @@ def _visit_FunctionDef(self, expr, **settings):
                             args_inout[i] = True
 
                         i_fa += 1
+                if (args_inout[i] is True and a.is_const is True) or (str(a) in all_assigned and a.is_const is True):
```

Comment:
What is the second condition checking for here? `is_const = True` and `a` is an array? Does this mean we cannot specify that arrays are constant?

Reply:
the second condition check if an argument with a `rank == 0` is modified so we raise an error if `const` is specified for this argument

Reply:
I also have a question here. If you are just evaluating `bool` variables, why not writing this?
```python
if (args_inout[i] and a.is_const) or (str(a) in all_assigned and a.is_const):
```
By the way, this is equivalent to
```python
if a.is_const and (args_inout[i] or str(a) in all_assigned):
```

Reply:
you are right the condition can be written in a simple and clear way